app/models/services/commitment_props.rb

```diff
@@ -0,0 +1,322 @@
+class Services::CommitmentProps
+  def initialize(commitment)
+    @commitment = commitment
+  end
+
+  def call
+    {
+      config:
+      {
+        action: @commitment.new_record? ? '/commitments.json' : "/commitments/#{ @commitments.id }.json",
+        method: @commitment.new_record? ? 'post' : 'put',
+        root_key: 'commitment'
+      },
+      survey: {
+        showPrevButton: false,
+        showQuestionNumbers: 'onPage',
+        showProgressBar: 'top',
+        requiredText: '',
+        pages: [
+          {
+            name: 'generalPage',
+            elements: [
```

Comment:
needs a hidden_field for criterium_id, which will already be on the record, passed in to the commitments controller via a params['criterium_id']